core/schema/host.go

```diff
@@ -5,60 +5,102 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/containerd/containerd/labels"
 	"github.com/dagger/dagger/core"
-	"github.com/dagger/dagger/core/socket"
+	"github.com/dagger/dagger/dagql"
+	"github.com/dagger/dagger/engine/sources/blob"
+	"github.com/opencontainers/go-digest"
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
 type hostSchema struct {
-	*APIServer
-
-	host *core.Host
-	svcs *core.Services
+	srv *dagql.Server
 }
 
 var _ SchemaResolvers = &hostSchema{}
 
-func (s *hostSchema) Name() string {
-	return "host"
-}
-
-func (s *hostSchema) Schema() string {
-	return Host
-}
-
-func (s *hostSchema) Resolvers() Resolvers {
-	return Resolvers{
-		"Query": ObjectResolver{
-			"host": PassthroughResolver,
-		},
-		"Host": ObjectResolver{
-			"directory":     ToResolver(s.directory),
-			"file":          ToResolver(s.file),
-			"unixSocket":    ToResolver(s.socket),
-			"setSecretFile": ToResolver(s.setSecretFile),
-			"tunnel":        ToResolver(s.tunnel),
-			"service":       ToResolver(s.service),
-		},
-	}
+func (s *hostSchema) Install() {
+	dagql.Fields[*core.Query]{
+		dagql.Func("host", func(ctx context.Context, parent *core.Query, args struct{}) (*core.Host, error) {
+			return parent.NewHost(), nil
+		}).Doc(`Queries the host environment.`),
+		dagql.Func("blob", func(ctx context.Context, parent *core.Query, args struct {
```

Comment:
I remember you mentioning this a while back, but I am forgetting and can't find where it's actually used now. Just wondering because the requirements around there being a lease on the blob are subtle, so was gonna double check all that.

Reply:
Sorry I pretty much immediately answered my own question after posting this comment, it's in the `srv.Select` of `LoadBlob` in `core/host.go`

Reply:
It's used via `core.LoadBlob`:

https://github.com/dagger/dagger/blob/adbcd9cb682b27b6d4848437db79fe08054a6163/core/host.go#L35-L61

Which is currently only called by `Host.Directory`:

https://github.com/dagger/dagger/blob/adbcd9cb682b27b6d4848437db79fe08054a6163/core/host.go#L63-L88

Which was probably hard to find because it's an example of an 'internal schema call' (don't know what to call it), which is what we do when we're interested in getting an ID-able value back rather than a raw core type. In this case it's so that `host.directory` returns a highly specific `biob()`-based ID instead or a `host.directory` ID, which is meaningless when passed between different contexts (i.e. Zenith modules).

Would definitely appreciate a double-check on the lease stuff :pray: